apps/widbattpwr/widget.ts

```diff
@@ -0,0 +1,71 @@
+(() => {
+	const intervalLow  = 60000;
+	const intervalHigh = 2000;
+	const width = 30;
+	const height = 24;
+
+	const powerColour = (pwr: number) =>
+		pwr >= 23000
+			? "#f00" // red, e.g. GPS ~20k
+			: pwr > 2000
+			? "#fc0" // yellow, e.g. CPU ~1k, HRM ~700
+			: "#0f0"; // green: ok
+
+	const drawBar = (x: number, y: number, batt: number) =>
+		g.fillRect(x+1, y+height-3, x+1+(width-2)*batt/100, y+height-1);
+
+	const drawString = (x: number, y: number, txt: string) =>
+		g.drawString(txt, x + 14, y + 10);
+
+	function draw(this: Widget) {
+		let x = this.x!;
+		let y = this.y!;
+
+		const batt = E.getBattery();
+		const pwr = E.getPowerUsage();
+		let usage = 0;
+		for(const key in pwr.device){
+			if(!/^(LCD|LED)/.test(key))
+				usage += pwr.device[key];
+		}
+		const pwrColour = powerColour(usage);
+
+		g.reset()
+			.setBgColor(g.theme.bg)
+			.clearRect(x, y, x + width - 1, y + height - 1);
+
+		g.setColor(g.theme.fg);
+		drawBar(x, y, 100);
+		g.setColor(pwrColour);
+		drawBar(x, y, batt);
+
+		g.setFontAlign(0, 0);
+		g.setFont("Vector", 16);
+		{
+			const hrs = 200000 / usage;
```

Comment:
@gfwilliams thinking about it - does this assume the battery is fully charged? Would the 200k need to be scaled down based on the battery's percentage?

Reply:
If it was time until flat then yes, I think so. Also battery is 175mAh so that needs a tweak anyway :)

Reply:
Cool, I'll linearly scale it based on the battery percentage. I guess we'll want to change this defininition in @Poolitzer's additions to `daisy` too?

https://github.com/espruino/BangleApps/blob/3211cf6955624ea01913581da005106189f1e4e5/apps/daisy/app.js#L119-L130

Reply:
Now I do feel though we should somehow export this that not all implementations of getPowerUsage() => remaining hours/days have to redo this, since it is math you can quite easily mess up a bit

Reply:
Sounds good - would you mind if I merge this as-is and did the consolidation of getPowerUsage --> hours/days in a separate PR? And yes - I got the 200k from that forum post too

Reply:
PR ready: #3498, gonna merge this